lms/djangoapps/etc_api/utils.py

```diff
@@ -0,0 +1,51 @@
+
+from django.conf import settings
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from student.forms import PasswordResetFormNoActive
+from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
+from rest_framework import permissions
+from django.http import Http404
+
+
+def send_activation_email(request):
+    form = PasswordResetFormNoActive(request.data)
+    if form.is_valid():
+        form.save(use_https=request.is_secure(),
+                  from_email=configuration_helpers.get_value(
+                      'email_from_address', settings.DEFAULT_FROM_EMAIL),
+                  request=request,
+                  subject_template_name='etc_api/set_password_subject.txt',
+                  email_template_name='etc_api/set_password_email.html'
+        )
+        return True
+    else:
+        return False
+
+class ApiKeyHeaderPermissionInToken(ApiKeyHeaderPermission, permissions.IsAuthenticated):
```

Comment:
Why don't you use ApiKeyHeaderPermission and create the new one?

Reply:
It's need for checking permission and checking enable or disable api for current site.